File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/how-to--add-an-updater-method_1.vb
    Public Shared Sub Update(ByVal contact As Contact)
        Const ServerName As String = "MySQLServerName"
        Dim dataContext As AdventureWorksDataContext = _
            New AdventureWorksDataContext("Data Source=" & ServerName & _
                ";Initial Catalog=AdventureWorks;Integrated Security=True")

        Dim ContactToUpdate As Contact = (From Contacts In dataContext.Contacts.AsEnumerable().Take(20) _
             Where Contacts.ContactID = contact.ContactID _
             Select Contacts).Single()

        With ContactToUpdate
            .FirstName = contact.FirstName
            .LastName = contact.LastName
            .EmailAddress = contact.EmailAddress
            .Phone = contact.Phone
            .EmailPromotion = contact.EmailPromotion
            .NameStyle = contact.NameStyle
            .PasswordHash = contact.PasswordHash
            .PasswordSalt = contact.PasswordSalt
        End With

        dataContext.SubmitChanges()

    End Sub

File: OLSource1/v1/VS_officedev/codesnippet/CSharp/how-to--add-an-updater-method_1.cs
        public static void Update(Contact contact)
        {
            const string ServerName = "MySQLServerName";
            AdventureWorksDataContext dataContext = new AdventureWorksDataContext
                  ("Data Source=" + ServerName + ";" +
                   "Initial Catalog=AdventureWorks;Integrated Security=True");
           
            var contactToUpdate = (from contacts in dataContext.Contacts
                                    where contacts.ContactID == contact.ContactID
                                    select contacts).Single();

            contactToUpdate.FirstName = contact.FirstName;
            contactToUpdate.LastName = contact.LastName;
            contactToUpdate.EmailAddress = contact.EmailAddress;
            contactToUpdate.Phone = contact.Phone;
            contactToUpdate.EmailPromotion = contact.EmailPromotion;
            contactToUpdate.NameStyle = contact.NameStyle;
            contactToUpdate.PasswordHash = contact.PasswordHash;
            contactToUpdate.PasswordSalt = contact.PasswordSalt;
            contactToUpdate.ModifiedDate = DateTime.Now;
            contactToUpdate.rowguid = Guid.NewGuid();
            dataContext.SubmitChanges();

        }